src/openms/source/KERNEL/ConsensusMap.cpp

```diff
@@ -730,6 +731,96 @@ OPENMS_THREAD_CRITICAL(oms_log)
     return true;
   }
 
+  void ConsensusMap::split(std::vector<FeatureMap>& fmaps, ConsensusMap::SplitMeta mode) const
+  {
+    // Prepare output vector
+    fmaps.clear();
+    Size numbr_exps = column_description_.size();
+    fmaps.resize(numbr_exps);
+
+    // Check for Isobaric Analyzer
+    bool iso_analyze = true;
+    auto cm_dp = this->getDataProcessing(); // get a copy to avoid calling .begin() and .end() on two different temporaries
+    if (all_of(cm_dp.begin(), cm_dp.end(), [](const OpenMS::DataProcessing& dp)
+                                           { return (dp.getProcessingActions().count(DataProcessing::QUANTITATION) == 0); }))
+    {
+      iso_analyze = false;
+    }
+
+    for (const auto& cf : *this)
+    {
+      // Create new Features from FeatureHandles
+      std::vector<BaseFeature> new_feats(numbr_exps);
```

Comment:
this might actually be a problem.
Not all CFs have basefeatures from all maps. (in IsobaricAnalyzer this may be the case, but not in general).

So you are probably better off to `push_back()` the basefeature into the correct featuremap right away and leave other featuremaps untouched.

Reply:
> Not all CFs have basefeatures from all maps. (in IsobaricAnalyzer this may be the case, but not in general).

This should'nt be a problem though, right? Because the basefeature is added corresponding to its map index.
`new_feats[fh.getMapIndex()] = feat;`

Reply:
its still a problem, because the vector has `numbr_expr` BF's, which are all pushed into their respective FeatureMaps later, even though they might just be default-constructed (and thus 'empty')

Reply:
you can go with a `map<int, BaseFeature>` instead of `vector` and instead of `new_feats[0]` (as used below), use `*new_feats.begin()`. That should work.